lib/tcpip/network_wrapper.cc

```diff
@@ -23,6 +25,46 @@ constexpr bool UseIPv6 = CHERIOT_RTOS_OPTION_IPv6;
 // IP thread global lock. See comment in FreeRTOS_IP_wrapper.c.
 extern struct FlagLockState ipThreadLockState;
 
+extern "C" int ipFOREVER(void)
+{
+	// We must interrupt the loop if a reset is ongoing (`Restarting`).
+	// However, we must still allow the loop to run during the early stages
+	// of a reset (`IpThreadKicked`).
+	uint32_t state = restartState.load();
+	return (state == 0) || ((state & IpThreadKicked) != 0);
```

Comment:
Maybe comment here that it's fine if the state changes after the test here because we'll be kicked again in the next loop iteration?